### PR TITLE
Closes issue #384: Option to Capture the photo using Volume Buttons

### PR DESCRIPTION
**Pull request by rishz**
Fixes issue #384: Option to Capture the photo using Volume Buttons

Changes: Added the option to capture images using Volume Buttons

Screenshots for the change:

### CODE REVIEW

**Comment by phimpme-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">:tada: Thanks for the PR @rishz we hope you have changed the variables in the upload_apk_forks.sh its for easy moderation.</td>
    </tr>
  </tbody>
</table>

:link: Maintainers: the apk to test can be found at https://github.com/rishz/phimpme-android/blob/apk/app-debug.apk

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by rishz** [comment]
@anantprsd5 Are you talking about the those two import statements? Yes, actually that was a result of [Code Styling of Import statements](https://source.android.com/source/code-style#order-import-statements) that Android Studio automatically does. :)

**Comment by Abhi2424shek** [comment]
You can easily prevent it and would it be nice that this option in settings?

**Comment by rishz** [comment]
@Abhi2424shek Added the option in settings. Check my recent commit. :)

screen121

File: app/src/main/java/vn/mbm/phimp/me/Settings.java [hunk 1]
```diff
@@ -459,6 +453,22 @@ public void onCheckedChanged(CompoundButton buttonView, boolean isChecked)
 			}
 		});
 
+		CheckBox checkVolumeBtn = (CheckBox)getView().findViewById(R.id.checkbox_volume_btn);
+		PhimpMe.check_volume_btn_to_capture = checkVolumeBtn.isChecked();
+		checkVolumeBtn.setOnCheckedChangeListener(new OnCheckedChangeListener()
+		{
```
Comment (Abhi2424shek) [inline app/src/main/java/vn/mbm/phimp/me/Settings.java]: Make this bracket go up to the previous line.

File: app/src/main/java/vn/mbm/phimp/me/Settings.java [hunk 2]
```diff
@@ -459,6 +453,22 @@ public void onCheckedChanged(CompoundButton buttonView, boolean isChecked)
...
+			@Override
+			public void onCheckedChanged(CompoundButton buttonView, boolean isChecked)
+			{
+				if(isChecked==true){
+					PhimpMe.check_volume_btn_to_capture=true;
```
Comment (Abhi2424shek) [inline app/src/main/java/vn/mbm/phimp/me/Settings.java]: Can you implement shared preferences so that it gets saved.

**Comment by rishz** [comment]
@Abhi2424shek Done. Check my recent commit.

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -639,6 +639,7 @@
     <string name="camera_settings">Camera Settings</string>
     <string name="volume_to_capture">Volume Button to Capture</string>
     <string name="volume_to_capture_detail">Enable/Disable Volume button to capture</string>
+    <string name="volumeBtnPrefs">volumeBtnPrefs</string>
```
Comment (Abhi2424shek) [inline app/src/main/res/values/strings.xml]: Make the string name to `camel_case` and a more logical value.

File: app/src/main/java/vn/mbm/phimp/me/PhimpMe.java [hunk 4]
```diff
@@ -805,6 +960,29 @@ public boolean onKeyDown(int keycode, KeyEvent event) {
...
+                                AudioManager.ADJUST_LOWER,
+                                AudioManager.FLAG_REMOVE_SOUND_AND_VIBRATE);
+                        preview.mCamera.takePicture(null, null, jpegCallback);
+                        return true;
+                    case KeyEvent.KEYCODE_FOCUS:
+                        preview.mCamera.takePicture(null, null, jpegCallback);
+                        return true;
+                    default:
+                        return false;
+//                Toast.makeText(ctx, "button pressed", Toast.LENGTH_SHORT).show();
```
Comment (Abhi2424shek) [inline app/src/main/java/vn/mbm/phimp/me/PhimpMe.java]: Remove this unused comment

File: app/src/main/java/vn/mbm/phimp/me/Settings.java [hunk 5]
```diff
@@ -790,6 +805,19 @@ public void run() {
...
+
+	@Override
+	public void onStop() {
+
+		boolean volumeBtnCheckVal = checkVolumeBtn.isChecked();
+		SharedPreferences sPrefs = getActivity().getPreferences(MODE_PRIVATE);
+		SharedPreferences.Editor editor = sPrefs.edit();
+		editor.putBoolean(getResources().getString(R.string.volumeBtnPrefs),volumeBtnCheckVal);
+		editor.commit();
+
```
Comment (Abhi2424shek) [inline app/src/main/java/vn/mbm/phimp/me/Settings.java]: Avoid unwanted blank lines.

**Comment by rishz** [comment]
@Abhi2424shek done that. Thanks for those suggestions. :)

**Comment by Abhi2424shek** [comment]
Thanks @rishz

**Pull request merged**